### PR TITLE
chore: add deploy job to test and deploy gh workflow

### PR DESCRIPTION
**Pull request by Hunga1**
Added a deploy job to the test and deploy GitHub Action workflow that creates a new GitHub release when run for tags. Updated build status badge and Slack notification to cover deployments by the workflow.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/sendgrid/smtpapi-php/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary documentation about the functionality in the appropriate .md file
- [x] I have added inline documentation to the code I modified

### CODE REVIEW

File: .github/workflows/test-and-deploy.yml [hunk 1]
```diff
@@ -44,19 +43,37 @@ jobs:
...
+    deploy:
+        name: Deploy
+        if: success() && github.ref_type == 'tag'
+        needs: [ test ]
+        runs-on: ubuntu-latest
+        steps:
+            -   name: Checkout smtpapi-php
+                uses: actions/checkout@v2
+
+            -   name: Build Release Artifacts
```
[the remark this answers is not in the document]
Comment (Hunga1) [reply .github/workflows/test-and-deploy.yml:55]: Should the build environment use the latest supported PHP version (7.4)?

File: .github/workflows/test-and-deploy.yml [hunk 2]
```diff
@@ -44,19 +43,37 @@ jobs:
 
             -   name: Run Tests
                 run: make test
+    deploy:
+        name: Deploy
+        if: success() && github.ref_type == 'tag'
+        needs: [ test ]
+        runs-on: ubuntu-latest
+        steps:
+            -   name: Checkout smtpapi-php
```
[the remark this answers is not in the document]
Comment (Hunga1) [reply .github/workflows/test-and-deploy.yml]: Done. Fixed alignment issues, and converted the file over to 2 space indentation

File: .github/workflows/test-and-deploy.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (childish-sambino) [reply .github/workflows/test-and-deploy.yml]: Yup, I think that's fine.

[the remark this answers is not in the document]
Comment (Hunga1) [reply .github/workflows/test-and-deploy.yml]: Done

**Review by childish-sambino — APPROVED** [review]

**Pull request merged**